tests/test_addons.py

```diff
@@ -0,0 +1,62 @@
+import unittest
+from typing import Any, Dict, Optional
+
+from scrapy.crawler import Crawler
+from scrapy.settings import BaseSettings
+from scrapy.utils.test import get_crawler
+
+
+class GoodAddon:
+    def __init__(self, config: Optional[Dict[str, Any]] = None) -> None:
```

Comment:
I'm not sure, what's the use case? It seems you can't use such addons with a custom config in setting.py, as a class needs to be used, not an instance, right?

Reply:
It looked good to me on my latest pass. Though I guess the `AddonTest` below only really tests BaseSettings.update, so maybe it is not that useful.

Reply:
I think that users shouldn't be defining addons with `__init__` arguments, unless the addon has a from_crawler method which passes these arguments to `__init__`.

Reply:
I don’t think it was meant as an example of a real use case, but as a shortcut for a clean test later on.

Reply:
It's just a remnant of old paradigm and it's handy to be able to use the same class with different settings, I guess I can rewrite it to a factory.